### PR TITLE
Add functions to replicate old HCL V1 Behavior for backwards compatability

### PR DESCRIPTION
**Pull request by sarahalsmiller**
- Added additional functions to replicate parsers old behavior to teams looking to switch aren't impacted by data that was already validated using the old parsers functionality. 
- Added additional test cases
- Cleaned up a segment of the duplicate code that was functionless

### CODE REVIEW

File: decoder.go [hunk 1]
```diff
@@ -35,7 +46,19 @@ func Unmarshal(bs []byte, v interface{}) error {
...
+
+// Decode reads the given input and decodes it into the structure
+// given by `out`.
+func DecodeErrorOnDuplicates(out interface{}, in string) error {
+	return decode(out, in, true)
+}
+
+// Decode reads the given input and decodes it into the structure
+// given by `out`.
+func decode(out interface{}, in string, errorOnDuplicateAtributes bool) error {
```
Comment (xwa153) [inline decoder.go:60]: very very nit: the function descriptions for `Decode` `DecodeErrorOnDuplicates` and `decode` are identical, maybe give a small difference to make it clearer.

File: decoder.go [hunk 2]
```diff
@@ -35,7 +46,19 @@ func Unmarshal(bs []byte, v interface{}) error {
 // Decode reads the given input and decodes it into the structure
 // given by `out`.
 func Decode(out interface{}, in string) error {
-	obj, err := Parse(in)
+	return decode(out, in, false)
+}
+
+// Decode reads the given input and decodes it into the structure
```
Comment (sarahalsmiller) [inline decoder.go:52]:
```suggestion
// DecodeErrorOnDuplicates reads the given input and decodes it into the structure but errrors on duplicate attribute key
```

File: decoder.go [hunk 3]
```diff
@@ -35,7 +46,19 @@ func Unmarshal(bs []byte, v interface{}) error {
...
+// given by `out`.
+func DecodeErrorOnDuplicates(out interface{}, in string) error {
+	return decode(out, in, true)
+}
+
+// Decode reads the given input and decodes it into the structure
```
Comment (sarahalsmiller) [inline decoder.go:58]:
```suggestion
// decode reads the given input and decodes it into the structure, takes in a boolean to determine if it should error on duplicate attribute
```

File: decoder.go [hunk 4]
```diff
@@ -35,7 +46,19 @@ func Unmarshal(bs []byte, v interface{}) error {
...
+}
+
+// DecodeErrorOnDuplicates reads the given input and decodes it into the structure but errrors on duplicate attribute key
+// given by `out`.
+func DecodeErrorOnDuplicates(out interface{}, in string) error {
+	return decode(out, in, true)
+}
+
+// decode reads the given input and decodes it into the structure, takes in a boolean to determine if it should error on duplicate attribute
+// given by `out`.
```
Comment (sarahalsmiller) [inline decoder.go:59]:
```suggestion
// decode reads the given input and decodes it into the structure given by `out`.
// takes in a boolean to determine if it should error on duplicate attribute
```

**Pull request merged**